c/parallel/include/cccl/c/types.h

```diff
@@ -20,7 +20,11 @@
 #  define CCCL_C_API __attribute__((visibility("default")))
 #endif // !_WIN32
 
-enum class cccl_type_enum
+#include <cccl/c/extern_c.h>
+
+CCCL_C_EXTERN_BEGIN
+
+enum cccl_type_enum
 {
   INT8    = 0,
```

Comment:
All enumerators need to have a common prefix (ex: `CCCL_TYPE_INT8` instead of `INT8`) because to C compilers all enumerators are considered global constants.

An additional consideration for internal codegen is: Add the enum type as a prefix to the enumerators. So to sum up this is preferred:
```C
typedef enum {
  CCCL_TYPE_INT8 = 0,
  CCCL_TYPE_INT16 = 1,
  ...
} ccclType;
```
(both `typedef` and camel case type names are optional)

Reply:
Pushed changes to rename `cccl_types_enum::INT8` to `cccl_types_enum::CCCL_INT8`, and so on. Also similarly renamed `cccl_op_kind_it::STATEFUL` to `cccl_op_kind_t::CCCL_STATEFUL`, and so on.

Reply:
Just to clarify, are we targeting C rather than C++ to support languages outside of Python?

Reply:
Yes, that is what I understood too.

Reply:
Yes. Among all benefits, C ABI is clean and simple.